transport-classes-io_uring/src/main/java/io/netty/channel/uring/IoUringBufferRingConfig.java

```diff
@@ -210,11 +228,15 @@ public Builder batchSize(int batchSize) {
         /**
          * Set the {@link IoUringBufferRingAllocator} to use to allocate {@link io.netty.buffer.ByteBuf}s.
          *
-         * @param allocator The allocator.
-         * @return          This builder.
+         * @param allocator         The allocator.
+         * @param batchAllocation   {@code true} if the ring should always be filled via a batch allocation or
+         *                          {@code false} if we will try to allocate a new {@link ByteBuf} as soon as
+         *                          as we used a buffer from the ring.
+         * @return                  This builder.
          */
-        public Builder allocator(IoUringBufferRingAllocator allocator) {
+        public Builder allocator(IoUringBufferRingAllocator allocator, boolean batchAllocation) {
```

Comment:
Any reason the `batchAllocation` is not a separate builder method?

Reply:
I can change if you want but it is related to the allocator so I thought we might just want to combine it

Reply:
@chrisvest https://github.com/netty/netty/pull/15479